Does not build: wrong argument in memcall.Protect: memcall.NoAccess... are functions

**Describe the bug**
A fresh go get of memguard results in vet errors.
# github.com/awnumar/memguard/core
src/github.com/awnumar/memguard/core/buffer.go:84:29: cannot use memcall.NoAccess (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect
src/github.com/awnumar/memguard/core/buffer.go:87:29: cannot use memcall.NoAccess (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect
src/github.com/awnumar/memguard/core/buffer.go:121:30: cannot use memcall.ReadOnly (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect
src/github.com/awnumar/memguard/core/buffer.go:142:30: cannot use memcall.ReadWrite (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect
src/github.com/awnumar/memguard/core/buffer.go:165:29: cannot use memcall.ReadWrite (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect
src/github.com/awnumar/memguard/core/exit.go:67:21: cannot use memcall.ReadWrite (type func() memcall.MemoryProtectionFlag) as type memcall.MemoryProtectionFlag in argument to memcall.Protect

**To Reproduce**
Steps to reproduce the behaviour:
go get github.com/awnumar/memguard

--- a/AUTHORS
+++ b/AUTHORS
@@ -7,3 +7,4 @@ Carlo Alberto Ferraris <cafxx@strayorange.com>
 dotcppfile <dotcppfile@gmail.com>
 Joseph Richey <joerichey@google.com>
 Neven Sajko <nsajko@gmail.com>
+Paul Zeinlinger <paul.zeinlinger@gmail.com>

--- a/core/buffer.go
+++ b/core/buffer.go
@@ -81,10 +81,10 @@ func NewBuffer(size int) (*Buffer, error) {
 	Copy(b.postguard, b.canary)
 
 	// Make the guard pages inaccessible.
-	if err := memcall.Protect(b.preguard, memcall.NoAccess); err != nil {
+	if err := memcall.Protect(b.preguard, memcall.NoAccess()); err != nil {
 		Panic(err)
 	}
-	if err := memcall.Protect(b.postguard, memcall.NoAccess); err != nil {
+	if err := memcall.Protect(b.postguard, memcall.NoAccess()); err != nil {
 		Panic(err)
 	}
 
@@ -118,7 +118,7 @@ func (b *Buffer) Freeze() {
 	// Only do anything if currently mutable.
 	if b.mutable {
 		// Make the memory immutable.
-		if err := memcall.Protect(b.inner, memcall.ReadOnly); err != nil {
+		if err := memcall.Protect(b.inner, memcall.ReadOnly()); err != nil {
 			Panic(err)
 		}
 		b.mutable = false
@@ -139,7 +139,7 @@ func (b *Buffer) Melt() {
 	// Only do anything if currently immutable.
 	if !b.mutable {
 		// Make the memory mutable.
-		if err := memcall.Protect(b.inner, memcall.ReadWrite); err != nil {
+		if err := memcall.Protect(b.inner, memcall.ReadWrite()); err != nil {
 			Panic(err)
 		}
 		b.mutable = true
@@ -162,7 +162,7 @@ func (b *Buffer) Destroy() {
 	}
 
 	// Make all of the memory readable and writable.
-	if err := memcall.Protect(b.memory, memcall.ReadWrite); err != nil {
+	if err := memcall.Protect(b.memory, memcall.ReadWrite()); err != nil {
 		Panic(err)
 	}
 

--- a/core/exit.go
+++ b/core/exit.go
@@ -64,7 +64,7 @@ func Panic(v interface{}) {
 	// Wipe all of the currently active LockedBuffers.
 	for _, b := range buffers.list {
 		if !b.mutable {
-			memcall.Protect(b.inner, memcall.ReadWrite)
+			memcall.Protect(b.inner, memcall.ReadWrite())
 		}
 		Wipe(b.Data())
 	}

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,6 @@ module github.com/awnumar/memguard
 go 1.12
 
 require (
-	github.com/awnumar/memcall v0.0.0-20190801095353-58a204f562ae
+	github.com/awnumar/memcall v0.0.0-20190808153224-92b5aac61e68
 	golang.org/x/crypto v0.0.0-20190701094942-4def268fd1a4
-	golang.org/x/sys v0.0.0-20190804053845-51ab0e2deafa // indirect
 )

